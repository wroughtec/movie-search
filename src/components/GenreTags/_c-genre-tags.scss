.c-genre-tags {
  display: flex;
  margin-bottom: .75rem;

  flex-wrap: wrap;
}

.c-genre-tags__tag {
  display: flex;
  height: 1.5rem;
  padding-right: .75rem;
  padding-left: .75rem;
  margin: .25rem;

  align-items: center;

  font-size: .65rem;
  white-space: nowrap;

  user-select: none;

  background-color: var(--tagBaseColour);
  border-radius: 1rem;
}
