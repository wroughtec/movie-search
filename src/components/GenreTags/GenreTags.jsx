// @flow
import React from 'react';
import './_c-genre-tags.scss';

type Props = {
  genreIds: number[],
  genres: GenresType[]
};

export const GenreTags = ({ genreIds, genres }: Props) => {
  const tags = [];

  let genreTags = 'No genres';

  if (genres && genres.length && genreIds && genreIds.length) {
    genreIds.forEach(genreId => tags.push(...genres.filter(obj => obj.id === genreId)));

    genreTags = tags.map(tag => {
      const { id, name } = tag;
      return (
        <span key={id} className={`c-genre-tags__tag c-genre-tags__tag--${id}`}>
          {name}
        </span>
      );
    });
  }

  return <div className="c-genre-tags">{genreTags}</div>;
};
