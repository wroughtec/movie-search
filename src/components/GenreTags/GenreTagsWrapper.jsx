// @flow

import React from 'react';
import { Consumer } from 'components/SearchContext/SearchContext';
import { GenreTags } from 'components/GenreTags/GenreTags';

type Props = {
  genreIds: number[]
};

export const GenreTagsWrapper = ({ genreIds }: Props) => (
  <Consumer>{({ genres }) => <GenreTags genres={genres} genreIds={genreIds} />}</Consumer>
);
