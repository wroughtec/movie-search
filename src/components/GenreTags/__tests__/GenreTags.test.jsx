// @flow

import React from 'react';
import { shallow, mount } from 'enzyme';
import { GenreTags } from 'components/GenreTags/GenreTags';
import { GenreTagsWrapper } from 'components/GenreTags/GenreTagsWrapper';

describe('GenreTags', () => {
  let component, props;
  const genres: GenresType[] = [
    {
      id: 28,
      name: 'Action'
    },
    {
      id: 12,
      name: 'Adventure'
    },
    {
      id: 878,
      name: 'Science Fiction'
    }
  ];

  beforeEach(() => {
    props = {
      genreIds: [12, 28, 878],
      genres
    };

    component = shallow(<GenreTags {...props} />);
  });

  test('Renders', () => {
    expect(component).toMatchSnapshot();
  });

  test('no genres', () => {
    const newComponent = shallow(<GenreTags genreIds={props.genreIds} genres={[]} />);

    expect(newComponent).toMatchSnapshot();
  });

  // @TODO look into better ways of testing the new Context API seems Enzymes support in not yet complete
  test('wrapper renders', () => {
    const wrapper = shallow(<GenreTagsWrapper genreIds={props.genreIds} />, { context: { genres } });

    expect(wrapper).toMatchSnapshot();
  });
});
