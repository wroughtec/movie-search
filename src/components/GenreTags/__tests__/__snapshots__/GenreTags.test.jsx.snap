// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreTags Renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GenreTags
    genreIds={
      Array [
        12,
        28,
        878,
      ]
    }
    genres={
      Array [
        Object {
          "id": 28,
          "name": "Action",
        },
        Object {
          "id": 12,
          "name": "Adventure",
        },
        Object {
          "id": 878,
          "name": "Science Fiction",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="c-genre-tags__tag c-genre-tags__tag--12"
        >
          Adventure
        </span>,
        <span
          className="c-genre-tags__tag c-genre-tags__tag--28"
        >
          Action
        </span>,
        <span
          className="c-genre-tags__tag c-genre-tags__tag--878"
        >
          Science Fiction
        </span>,
      ],
      "className": "c-genre-tags",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "12",
        "nodeType": "host",
        "props": Object {
          "children": "Adventure",
          "className": "c-genre-tags__tag c-genre-tags__tag--12",
        },
        "ref": null,
        "rendered": "Adventure",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "28",
        "nodeType": "host",
        "props": Object {
          "children": "Action",
          "className": "c-genre-tags__tag c-genre-tags__tag--28",
        },
        "ref": null,
        "rendered": "Action",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "878",
        "nodeType": "host",
        "props": Object {
          "children": "Science Fiction",
          "className": "c-genre-tags__tag c-genre-tags__tag--878",
        },
        "ref": null,
        "rendered": "Science Fiction",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="c-genre-tags__tag c-genre-tags__tag--12"
          >
            Adventure
          </span>,
          <span
            className="c-genre-tags__tag c-genre-tags__tag--28"
          >
            Action
          </span>,
          <span
            className="c-genre-tags__tag c-genre-tags__tag--878"
          >
            Science Fiction
          </span>,
        ],
        "className": "c-genre-tags",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "12",
          "nodeType": "host",
          "props": Object {
            "children": "Adventure",
            "className": "c-genre-tags__tag c-genre-tags__tag--12",
          },
          "ref": null,
          "rendered": "Adventure",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "28",
          "nodeType": "host",
          "props": Object {
            "children": "Action",
            "className": "c-genre-tags__tag c-genre-tags__tag--28",
          },
          "ref": null,
          "rendered": "Action",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "878",
          "nodeType": "host",
          "props": Object {
            "children": "Science Fiction",
            "className": "c-genre-tags__tag c-genre-tags__tag--878",
          },
          "ref": null,
          "rendered": "Science Fiction",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`GenreTags no genres 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GenreTags
    genreIds={
      Array [
        12,
        28,
        878,
      ]
    }
    genres={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "No genres",
      "className": "c-genre-tags",
    },
    "ref": null,
    "rendered": "No genres",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No genres",
        "className": "c-genre-tags",
      },
      "ref": null,
      "rendered": "No genres",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`GenreTags wrapper renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GenreTagsWrapper
    genreIds={
      Array [
        12,
        28,
        878,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "Consumer": [Circular],
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": Object {
        "genres": Array [],
        "handleMovieSubmit": [Function],
        "handleSearchChange": [Function],
        "imageBaseUrl": "",
        "lg": "",
        "loading": true,
        "md": "",
        "popularPageNo": 1,
        "popularResults": Object {},
        "searchPageNo": 1,
        "searchResults": Object {},
        "searchTerms": "",
        "sm": "",
        "updateSearchTerms": [Function],
        "xl": "",
      },
      "_currentValue2": Object {
        "genres": Array [],
        "handleMovieSubmit": [Function],
        "handleSearchChange": [Function],
        "imageBaseUrl": "",
        "lg": "",
        "loading": true,
        "md": "",
        "popularPageNo": 1,
        "popularResults": Object {},
        "searchPageNo": 1,
        "searchResults": Object {},
        "searchTerms": "",
        "sm": "",
        "updateSearchTerms": [Function],
        "xl": "",
      },
      "unstable_read": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "genres": Array [],
          "handleMovieSubmit": [Function],
          "handleSearchChange": [Function],
          "imageBaseUrl": "",
          "lg": "",
          "loading": true,
          "md": "",
          "popularPageNo": 1,
          "popularResults": Object {},
          "searchPageNo": 1,
          "searchResults": Object {},
          "searchTerms": "",
          "sm": "",
          "updateSearchTerms": [Function],
          "xl": "",
        },
        "_currentValue2": Object {
          "genres": Array [],
          "handleMovieSubmit": [Function],
          "handleSearchChange": [Function],
          "imageBaseUrl": "",
          "lg": "",
          "loading": true,
          "md": "",
          "popularPageNo": 1,
          "popularResults": Object {},
          "searchPageNo": 1,
          "searchResults": Object {},
          "searchTerms": "",
          "sm": "",
          "updateSearchTerms": [Function],
          "xl": "",
        },
        "unstable_read": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "genres": Array [
        Object {
          "id": 28,
          "name": "Action",
        },
        Object {
          "id": 12,
          "name": "Adventure",
        },
        Object {
          "id": 878,
          "name": "Science Fiction",
        },
      ],
    },
    "hydrateIn": undefined,
  },
}
`;
